### PR TITLE
Detect JSON files during pull

### PR DESCRIPTION
**Pull request by mbtools**
Fixes JSON file detection (#5305)

### CODE REVIEW

**Comment by albertmink** [comment]
Pull results in
image
It is not quite there yet.

**Comment by albertmink** [comment]
Might be somewhere around here where there is a hard-coded reference to file ending `XML`.
https://github.com/abapGit/abapGit/blob/main/src/objects/zcl_abapgit_objects.clas.abap#L676

The method `read_xml` does what it is supposed to do 😄 
image

**Comment by mbtools** [comment]
this will take several steps to refactor since i can't test it myself...

**Comment by albertmink** [comment]
Is it necessary to run an ABAP System with support for CHKO object? It felt like this part of deserialization in AbapGit is not related to specific CHKO API.

**Comment by mbtools** [comment]
it isn't related. but on lower releases AG will filter object types that are not supported before you even get to this point.

you can see this on "Utilities > Debug Info":

image

The new AFF object types are not shown on my system.

Anyway, I think I got it with the next commit

**Comment by larshp** [comment]
there is also relevant metadata in the json files, so this needs refactoring sometime, if it works

**Comment by mbtools** [comment]
absolutely. it needs to be properly modeled (see #4861)

**Comment by mbtools** [comment]
@albertmink give it a try again

**Comment by albertmink** [comment]
Thanks for your immediate support.
Now, it is the missing devclass
image

File: src/objects/zcl_abapgit_objects.clas.abap [hunk 1]
```diff
@@ -656,9 +657,10 @@ CLASS zcl_abapgit_objects IMPLEMENTATION.
             lv_path = <ls_result>-path.
           ENDIF.
 
+          ls_item-devclass = lv_package.
```
Comment (larshp) [inline src/objects/zcl_abapgit_objects.clas.abap:660]: hmm, moving this statement seems like a hack, I guess its not the way the xml gets the package

**Comment by mbtools** [comment]
It's kind of a mistake in your deserialize method where it uses ms_item-devclass instead of iv_package. I made a small change so ms_item-devclass is set to the same package. 

FYI, I see you check the object type in several places. That's not necessary since the class will only be called if the obj_type matches the classname.  You can remove all of these and probably won't need to redefine the methods much:

```abap
    IF ms_item-obj_type <> 'CHKO'.
      RETURN.
    ENDIF.
```

File: src/objects/zcl_abapgit_objects.clas.abap (hunk 1 above)
Comment (mbtools) [reply src/objects/zcl_abapgit_objects.clas.abap:660]: the ls_item is passed to the constructor (ms_item) which includes devclass. it's confusing if this does not match the package passed to deserialize method. 🤷

Comment (mbtools) [reply src/objects/zcl_abapgit_objects.clas.abap:660]: https://github.com/abapGit/abapGit/blob/main/src/objects/aff/zcl_abapgit_object_common_aff.clas.abap#L153

**Comment by albertmink** [comment]
👋 No error message after pulling, so this might work for the JSON files.
Unlucky, that we didn't tested this behavior when working on #5163. Relaying on unit test was not enough.

**Comment by albertmink** [comment]
It is working as expected for CHKC and CHKO.
I could deserialize them into the ABAP system, requires #5320 #5319

**Pull request merged**